### DISCUSSION TITLE
Create a `coJustAwait` function similar to `coJustRun`

### DISCUSSION THREAD

**Discussion by SimonMarquis** [mockk/mockk#924] [Ideas]
👋  Hi,
Would you be open to add a `coJustAwait`, similar to `coJustRun`?

This function would simply stub the given block to wait indefinitely (until cancellation).
It could be used to mockk a suspend function to never return.

This can be especially usefull to test intermediate states, without having to explicitely specify coroutine dispatchers, and to prevent early returns.

**Comment by Raibaz** [comment]
Generally speaking, I'd say this makes sense.

Can you provide a code example of how you would expect this to work, though?

**Comment by SimonMarquis** [comment]
```kotlin
/**
 * Stub block to never return.
 *
 * Used to define what behaviour is going to be mocked.
 * @see [io.mockk.coJustRun]
 */
fun coJustAwait(
    stubBlock: suspend MockKMatcherScope.() -> Unit,
) = coEvery(stubBlock) coAnswers {
    awaitCancellation()
}
```

**Comment by SimonMarquis** [comment]
In the meantime, here is what it would look like in the code base https://github.com/mockk/mockk/pull/927

**Comment by SimonMarquis** [comment]
The PR has been merged, this discussion can be closed as answered.
